ros2cli/ros2cli/daemon/__init__.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ros2cli.node.direct import DirectNode
+from xmlrpc.server import SimpleXMLRPCServer
+
+PORT = 11211
+
+
+def main(*, script_name='_ros2_daemon', argv=None):
+    addr = ('localhost', PORT)
+    with DirectNode({}) as node:
+        server = SimpleXMLRPCServer(addr, logRequests=False)
+
+        try:
+            server.register_introspection_functions()
+
+            # expose getter functions of node
+            server.register_function(
+                _print_invoked_function_name(node.get_node_names))
+            server.register_function(
+                _print_invoked_function_name(node.get_topic_names_and_types))
+
+            # function to shutdown daemon remotely
+            shutdown = False
+
+            def shutdown_handler():
+                nonlocal shutdown
+                print('Remote shutdown requested')
+                shutdown = True
+            server.register_function(shutdown_handler, 'system.shutdown')
```

Comment:
So again, when does the daemon shutdown, if it is forked off ("spawned")? Or how does the user shut it down?